decoder/decoder.go

```diff
@@ -195,6 +195,55 @@ func forEachAttributeSchema() *schema.AttributeSchema {
 	}
 }
 
+func dynamicBlockSchema() *schema.BlockSchema {
+	return &schema.BlockSchema{
+		Description: lang.Markdown("A dynamic block to produce blocks dynamically by iterating over a given complex value"),
+		Type:        schema.BlockTypeMap,
+		Labels: []*schema.LabelSchema{
+			{Name: "name"},
+		},
+		Body: &schema.BodySchema{
+			Attributes: map[string]*schema.AttributeSchema{
+				"for_each": {
+					Expr: schema.ExprConstraints{
+						schema.TraversalExpr{OfType: cty.Map(cty.DynamicPseudoType)},
+						schema.TraversalExpr{OfType: cty.Set(cty.String)},
+						schema.LiteralTypeExpr{Type: cty.Map(cty.DynamicPseudoType)},
+						schema.LiteralTypeExpr{Type: cty.Set(cty.String)},
+					},
+					IsRequired: true,
+					Description: lang.Markdown("A meta-argument that accepts a map or a set of strings, and creates an instance for each item in that map or set.\n\n" +
+						"**Note**: A given block cannot use both `count` and `for_each`."),
+				},
+				"iterator": {
+					Expr:       schema.LiteralTypeOnly(cty.String),
+					IsOptional: true,
+					Description: lang.Markdown("The name of a temporary variable that represents the current " +
+						"element of the complex value. Defaults to the label of the dynamic block."),
+				},
+				"labels": {
+					Expr: schema.ExprConstraints{
+						schema.ListExpr{
+							Elem: schema.ExprConstraints{
+								schema.LiteralTypeExpr{Type: cty.String},
+								schema.TraversalExpr{OfType: cty.String},
+							},
+						},
+					},
+					IsOptional: true,
+					Description: lang.Markdown("A list of strings that specifies the block labels, " +
+						"in order, to use for each generated block."),
+				},
+			},
+			Blocks: map[string]*schema.BlockSchema{
+				"content": {
+					Description: lang.PlainText("The body of each generated block"),
```

Comment:
```suggestion
					Description: lang.PlainText("The body of each generated block"),
					MaxItems:    1,
```

We can set `MaxItems` here, so autocompletion only suggests a single `content` block.